### DISCUSSION TITLE
[development question]

### DISCUSSION THREAD

**Discussion by IvanIsak2000** [Spaceblock/cosmoterm#8] [General]
@mb6ockatf 
Ok, I've created empty `html` pages for user login and registration and added them to `app.py` what should I do with them - run tests or push? If push , then how do I document it? I remember you said something about `CHANGELOG`, but did not fully understand what to write there)

**Comment by mb6ockatf** [comment]
testing is all about python code. if there is a new code, then it should be tested (you can read about unittest yourself)

before commiting anything to git, describe your changes in changelog file. it also really helps with version tracking. you can read about changelog [here](https://keepachangelog.com/en/1.0.0/)

**Comment by mb6ockatf** [comment]
i used unittest framework [here](https://github.com/mb6ockatf/IVPT/tree/master), in tests folder - tests for database connection

**Reply by IvanIsak2000** [reply]
If we test the code, then fast api has a special asynchronous client https://fastapi.tiangolo.com/tutorial/testing/#using-testclient

**Reply by mb6ockatf** [reply]
great! we are using fastapi, not flask then - we don't need flask for such a messager

**Comment by mb6ockatf** [comment]
@IvanIsak2000 i removed `wiki` and `actions` tabs from the top bar. It's annoying for me when there're too many buttons. you may turn them on in the settings if you need them

**Comment by IvanIsak2000** [comment]
@mb6ockatf What you think about main framework? I want to try use `django` or should we leave flask? Waiting your opinion

**Reply by IvanIsak2000** [reply]
And `postgres` db will as main, because I learned a small how to create and connect that

**Reply by mb6ockatf** [reply]
@IvanIsak2000 postgres will do. i know none of django, so lets do flask.
i wrote a small makefile, so run `make build` to prepare the project, and just `make` to launch the whole thing

**Reply by mb6ockatf** [reply]
it's in the _improve_design_ branch currently

**Comment by mb6ockatf** [comment]
@IvanIsak2000 there were no git tags created. what are you writing to CHANGELOG, man?

**Comment by mb6ockatf** [comment]
@IvanIsak2000 we can use issues as todo list

**Comment by mb6ockatf** [comment]
By the time way, you do not like my design? You mentioned you want it to be modern

I started using sass, so you compile bootstrap and custom rules together with sass tool before starting application. There's an extension in vs code that watches sass files and automerges them

**Reply by IvanIsak2000** [reply]
Yeah, I got that:

![image](https://github.com/Spaceblock/cosmoterm/assets/79650307/90a47f19-69b1-4c4c-8110-d2f2b62b661e)
 Are you mean that?

**Reply by mb6ockatf** [reply]
@IvanIsak2000 oh, ok. If you don't like it, you may improve it. I'm not an artist, and what i did was quite satisfying for me. At least, it is much easier to maintain, than css file generated by some sort of Figma

**Reply by mb6ockatf** [reply]
You can change colors and font, if you like, and it'll look ok, i guess

Also, advise you Colors app – it's pallet generator

**Reply by mb6ockatf** [reply]
![Screenshot_20231119_204417_Finder](https://github.com/Spaceblock/cosmoterm/assets/104227451/95338447-ac1d-4228-a4c1-fcecdead7748)
That's the logo

**Reply by IvanIsak2000** [reply]
Yes, I know this app. As for me, as new to the bootstrap, I use standard themes. But I improve my skills
